### PR TITLE
detect vulnerable class in jar files

### PR DESCRIPTION
**Pull request by AdrianDiemerDev**

### CODE REVIEW

File: src/main/java/com/intrafind/log4shell/Log4shellUpdate.java [hunk 1]
```diff
@@ -100,6 +108,17 @@ private static void handleIfOldLog4j(Path path, Map<Path, String> toReplace, Lis
                 .ifPresent(replacementResource -> toReplace.put(path, replacementResource));
           }
         }
+        if (filename.endsWith(".jar") && replacements.keySet().stream().noneMatch(filename::startsWith)) {
+          try {
+            try (final InputStream fileInputStream = Files.newInputStream(path)) {
```
Comment (michael-kuss) [inline src/main/java/com/intrafind/log4shell/Log4shellUpdate.java]: I would use a BufferedStream on jar file parsing

File: src/main/java/com/intrafind/log4shell/Log4shellUpdate.java [hunk 2]
```diff
@@ -142,6 +161,22 @@ private static void delete(Path oldFile, List<Path> backups) {
     }
   }
 
+  private static boolean zipContainsLog4j(InputStream inputStream) throws IOException {
+    try (final ZipInputStream zipInputStream = new ZipInputStream(inputStream)) {
+      for (ZipEntry entry = zipInputStream.getNextEntry(); entry != null; entry = zipInputStream.getNextEntry()) {
+        if ("org/apache/logging/log4j/core/lookup/JndiLookup.class".equals(entry.getName())) {
+          return true;
```
Comment (michael-kuss) [inline src/main/java/com/intrafind/log4shell/Log4shellUpdate.java:170]: What if more than one jar/class is found in one zip?

**Review by michael-kuss — COMMENTED** [review]
Apart from the 2 comments looks fine to me

File: src/main/java/com/intrafind/log4shell/Log4shellUpdate.java (hunk 1 above)
Comment (AdrianDiemerDev) [reply src/main/java/com/intrafind/log4shell/Log4shellUpdate.java]: done

File: src/main/java/com/intrafind/log4shell/Log4shellUpdate.java (hunk 2 above)
Comment (AdrianDiemerDev) [reply src/main/java/com/intrafind/log4shell/Log4shellUpdate.java:170]: That can only happen with a broken jar.

Also, if the jar in question is patched an the user misses one of the duplicate classes, the tool will still detect the remaining class.

**Pull request merged**